### PR TITLE
fix: styles for the schedule component, make it mobile-friendly

### PR DESCRIPTION
**Pull request by helgastogova**
## What does this PR do?

- Fixes CAL-3797, Make the Schedule component mobile-friendly.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

Visually check the Schedule component view on the /availability/ page.

before (wide screen)
image

before (mobile view)
image

after  (wide screen)
image

after (mobile view)
image

design check (compare with Figma)
image

### CODE REVIEW

File: packages/features/schedules/components/Schedule.tsx
Comment (Ryukemeister) [inline packages/features/schedules/components/Schedule.tsx:107]: How about we move this `watchDayRange.length > 0` check one level higher. So instead of adding another conditional inside of the dayRanges component maybe something like this 
`!!watchDayRange ? Day ranges component : Skeleton component`

Comment (Ryukemeister) [reply packages/features/schedules/components/Schedule.tsx:107]: The `!!` will convert the `watchDayRange` variable to its boolean equivalent, similar to what `watchDayRange.length > 0` does. Or you can use `Boolean(watchDayRange)` but thats just a personal preference since both of them are the same

Comment (helgastogova) [reply packages/features/schedules/components/Schedule.tsx:107]: Hey @Ryukemeister, yes, I agree with you, it really does seem like it needs to be written exactly like this.
```
{!!watchDayRange ? (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
```
     
but if you try to do that, you will see that in that case, we have one extra empty div

image

the same will be if I use `!!watchDayRange` instead of `watchDayRange.length > 0` 

The difference between these checks in this case:

1. watchDayRange.length > 0 
This line checks if the length of the watchDayRange array is greater than zero. This means there must be at least one element in the array for the condition to return true.
2. !!watchDayRange
This line first converts watchDayRange to a boolean value. If watchDayRange is an object or array (even if empty), !!watchDayRange will always return true because objects and arrays are "truthy" values in JavaScript. Thus, this condition checks for the existence of the array, not its emptiness.
The difference is that the first condition checks for the presence of elements within the array, while the second checks for the existence of the array itself, regardless of its content. If watchDayRange is an empty array ([]), the first condition will return false, while the second will return true.


And if I do:
```
     {watchDayRange.length > 0 ? (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
```
I will see a skeleton when I don't need to see it
image

So the best option here is to leave it as it is

Comment (Ryukemeister) [reply packages/features/schedules/components/Schedule.tsx:107]: What if we do `watchDayRange && watchDayRange.length > 0`? That way day ranges only renders when watch day range is defined and its not empty. So what about  `watchDayRange && watchDayRange.length > 0 ? < DayRanges/> : < Skeleton/>`

Comment (helgastogova) [reply packages/features/schedules/components/Schedule.tsx:107]: `watchDayRange && watchDayRange.length > 0 `
will give us `false` if we have `[] `and we will see a <Skeleton /> when we don't want to see it

so, code might look like that
```
    <>
        {!watchDayRange && <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />}
        {watchDayRange.length > 0 && (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        )}
      </>
      ```
      
      or like that
```

 <>
        {watchDayRange ? (
          <>
            {watchDayRange.length > 0 && (
              <div className="flex sm:gap-2">
                <DayRanges
                  userTimeFormat={userTimeFormat}
                  labels={labels}
                  control={control}
                  name={name}
                  disabled={disabled}
                  className={{
                    dayRanges: className?.dayRanges,
                    timeRangeField: className?.timeRangeField,
                  }}
                />
                {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
              </div>
            )}
          </>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
      </>

```
  but not like that
```

        <>
        {watchDayRange && watchDayRange.length > 0 ? (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
      </>

```

**Review by sean-brydon — APPROVED** [review]
I'm happy with this thanks @helgastogova :) 

@Ryukemeister - I'll wait for your approval before we merge as you left some previous comments.

**Review by Ryukemeister — APPROVED** [review]

**Comment by shirazdole** [comment]
wow this sick!

**Pull request merged**
